packages/flutter/lib/src/widgets/platform_menu_bar.dart

```diff
@@ -0,0 +1,886 @@
+  ///
+  /// If the menu is disabled, then it can't be opened.
+  final bool enabled;
+
+  /// The callback that is called when this menu is opened.
+  final VoidCallback? onOpen;
+
+  /// The callback that is called when this menu is closed.
+  final VoidCallback? onClose;
+
+  /// The menu items in the submenu opened by this menu item.
+  ///
+  /// If empty, this menu item will act as if [enabled] were false.
+  final List<MenuItem> children;
+
+  /// Returns all descendant [MenuItem]s of this item.
+  List<MenuItem> get descendants => getDescendants(this);
+
+  /// Returns all descendants of the given item.
+  ///
+  /// This API is supplied so that implementers of [PlatformMenu] can share
+  /// this implementation.
+  static List<MenuItem> getDescendants(PlatformMenu item) {
+    return <MenuItem>[
+      for (final MenuItem child in item.children) ...<MenuItem>[
+        child,
+        if (child is PlatformMenu) ...child.descendants,
+      ],
+    ];
+  }
+
+  @override
+  Iterable<Map<String, Object?>> toChannelRepresentation(
+    DefaultPlatformMenuDelegate delegate, {
+    required int index,
+    required int count,
+    required DefaultPlatformMenuDelegateSerializableIdGenerator getId,
+  }) {
+    return <Map<String, Object?>>[serialize(this, delegate, getId)];
+  }
+
+  /// Converts the supplied object to the correct channel representation for the
+  /// 'flutter/menu' channel.
+  ///
+  /// This API is supplied so that implementers of [PlatformMenu] can share
+  /// this implementation.
+  static Map<String, Object?> serialize(
+    PlatformMenu item,
+    DefaultPlatformMenuDelegate delegate,
+    DefaultPlatformMenuDelegateSerializableIdGenerator getId,
+  ) {
+    final List<Map<String, Object?>> result = <Map<String, Object?>>[];
+    int index = 0;
+    for (final MenuItem childItem in item.children) {
+      if (childItem is! DefaultPlatformMenuDelegateSerializable) {
+        throw UnimplementedError('Tried to serialize a menu item that was not a DefaultPlatformMenuDelegateSerializer');
+      }
+      result.addAll((childItem as DefaultPlatformMenuDelegateSerializable).toChannelRepresentation(
+        delegate,
+        index: index,
+        count: item.children.length,
+        getId: getId,
+      ));
+      index += 1;
+    }
+    return <String, Object?>{
+      _kIdKey: getId(item),
+      _kLabelKey: item.label,
+      _kEnabledKey: item.enabled,
+      _kChildrenKey: result,
+    };
+  }
+
+  @override
+  List<DiagnosticsNode> debugDescribeChildren() {
+    return children.map<DiagnosticsNode>((MenuItem child) => child.toDiagnosticsNode()).toList();
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(StringProperty('label', label));
+    properties.add(FlagProperty('enabled', value: enabled, ifFalse: 'DISABLED'));
+  }
+}
+
+/// An interface for [MenuItem]s that group other menu items into sections
```

Comment:
nit: it's more than just an interface, no? You can and are supposed to instantiate this directly?